app/src/main/java/nl/tudelft/bbw/controller/KeyReader.java

```diff
@@ -81,20 +82,20 @@ public static EdDSAPrivateKey readPrivateKey() throws InvalidKeySpecException, I
      *
      * @return public key
      */
-    public static final EdDSAPublicKey readPublicKey() throws InvalidKeySpecException, IOException {
+    public static EdDSAPublicKey readPublicKey(String password) throws InvalidKeySpecException, IOException {
         final String publicKeyPath = "public.key";
-        final byte[] encodedPublicKey = readKey(publicKeyPath);
+        final byte[] encodedPublicKey = readKey(publicKeyPath, password);
         return convertToPublicKey(encodedPublicKey);
     }
 
     /**
-     * readPublicKey method
+     * StringPublicKey method
```

Comment:
Since this is the case in a lot of classes and functions in our code, and we have to work consistently, this should not be changed or should be addressed in another pull request.